zyte_api/aio/client.py

```diff
@@ -43,6 +51,38 @@ def _post_func(session):
         return session.post
 
 
+class ExtractResult(Mapping):
```

Comment:
I think we could directly inherit from `dict` like:

```python
class ExtractResult(dict):
    def __init__(self, *args, **kwargs):
        super().__init__(*args, **kwargs)
        self._http_response_body: Union[bytes|_NotLoaded] = _NOT_LOADED
    
ExtractResult(response)
```

Reply:
I believe [subclassing `dict` can be problematic](https://treyhunner.com/2019/04/why-you-shouldnt-inherit-from-list-and-dict-in-python/). Implementing `Mapping` also has the benefit(?) of making the class read-only (but only for top keys).

Reply:
I see your point. 👍 Yep, let's continue using `collections.abc.Mapping` to ensure the correct behavior. Immutability is also a good bonus.